lib/rubocop/cop/rails/require_dependency.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for the usage of `require_dependency`.
+      # Applications running in zeitwerk mode should not use require_dependency
```

Comment:
Can you add an explanation for this one? The following is an example:

```suggestion
      # This method is obsolete in zeitwerk mode. In zeitwerk mode semantics match Ruby's and
      # you do not need to be defensive with load order. Just refer to classes and modules normally.
      # If the constant name is dynamic, camelize if needed, and constantize.
      #
      # Applications running in zeitwerk mode should not use `require_dependency`.
      #
      # NOTE: This cop is disabled by default. Please enable it if you are using zeitwerk mode.
```